lib/core/uploaderError.js

```diff
@@ -17,29 +17,43 @@
 
 var sundial = require('sundial');
 
-// Create a new object, that prototypally inherits from the Error constructor.
-function UploaderError(message, stage, error) {
-
-  this.name = this.constructor.name;
-  this.message = message;
+// if another error is passed in then attach it
```

Comment:
Ah, I think this is the cause of some of the redundancy I complained about in the Trello card. Want to elaborate on why you think this is useful?